claim_batch/services.py

```diff
@@ -431,6 +432,109 @@ def _get_capitation_region_and_district(location_id):
 def do_process_batch(audit_user_id, location_id, period, year):
     processed_ids = set()  # As we update claims, we add the claims not in relative pricing and then update the status
     logger.debug("do_process_batch location %s for %s/%s", location_id, period, year)
+    batch_run = new BatchRun
+
+    # 0 prepare the batch run :  does it really make sense per location ? (Ideally per pool but the notion doesn't exist yet)
+    # 0.1 get all product concerned, all product that have are configured for the location
+    _, days_in_month = calendar.monthrange(year, period)
+    end_date = datetime.date(year, period, days_in_month)
+    period_quarter = period - 2 if period|divisibleby:4 else 0
+    period_sem = period - 5 if period|divisibleby:6 else 0
+
+    products = Product.objects\
+            .filter(validity_from__lte = start_date)\
+            .filter(validity_from__gte = end_date)\
+            .filter(validity_to__isnull = True)\
+            .filter(date_from__lte = end_date)\
+            .filter(Q(date_to__gte = start_date) | Q(date_to__isnull = True))\
+            .filter(location_id = location_id)
+    # 1 per product (Ideally per pool but the notion doesn't exist yet)
+    if products:
+        for product in products:
+
+            if product.period_rel_prices = 'Y' or product.period_rel_prices_ip = 'Y' or product.period_rel_prices_op = 'Y':
+                start_date = datetime.date(year, 1, 1)
+            else if product.period_rel_prices = 'S' or product.period_rel_prices_ip = 'S' or product.period_rel_prices_op = 'S':    
+                start_date = datetime.date(year, period_sem, 1)
+            else if product.period_rel_prices = 'Q' or product.period_rel_prices_ip = 'Q' or product.period_rel_prices_op = 'Q':    
+                start_date = datetime.date(year, period_quarter, 1)
+            else:
+                start_date = datetime.date(year, period, 1)
+
+            work_data.product = product
+            # 1.2 get all the payment plan per product
+            work_data.paymentplans = PaymentPlan.objects\
+                .filter(valid_to__lte = start_date)\
+                .filter(valid_from__gte = end_date)\
+                .filter(product = product)
+            # 1.3 retrive all the Item & service per product
+            work_data.items  = ClaimItem.objects\
+                .filter(validity_from__lte = start_date)\
```

Comment:
@delcroip Is this statement correct with 'validity_from__lte' and validity_from__gte'? The another ones looks fine. But I have concerns about this one.

Reply:
you are right use validity_from__gte but actually it this the processed date of the claim that should be used remove the validity_from altogether

Reply:
for item, service contribution and claims

Reply:
ok, noted.